### PR TITLE
[FOGL-7049]: Changed timestamp to show year, month and date along with time in tabular data

### PR DESCRIPTION
**Pull request by monika-Dianomic**

### CODE REVIEW

File: src/app/components/core/asset-readings/readings-graph/readings-graph.component.ts [hunk 1]
```diff
@@ -622,7 +622,7 @@ export class ReadingsGraphComponent implements OnDestroy {
           type: 'time',
           time: {
             unit: 'second',
-            tooltipFormat: 'HH:mm:ss:SSS',
+            tooltipFormat: 'YYYY-MM-DD HH:mm:ss.SSS',
             displayFormats: {
               unit: 'second',
               second: 'HH:mm:ss'
```
Comment (praveen-garg) [inline src/app/components/core/asset-readings/readings-graph/readings-graph.component.ts:628]: what is the use of displayFormats second item? if this is for X axis then for > 8 hours should we show YYYY-MM-DD as well?

File: src/app/components/core/asset-readings/readings-graph/readings-graph.component.ts [hunk 2]
```diff
@@ -436,7 +436,7 @@ export class ReadingsGraphComponent implements OnDestroy {
             imageReadings.push({
               datapoint: k,
               imageData: value,
-              timestamp: this.dateFormatter.transform(r.timestamp, 'HH:mm:ss')
+              timestamp: this.dateFormatter.transform(r.timestamp, 'HH:mm:ss.SSS')
```
Comment (mshariq-nerd) [inline src/app/components/core/asset-readings/readings-graph/readings-graph.component.ts:439]: Missing `YYYY-MM-DD` with time format

File: src/app/components/core/asset-readings/readings-graph/readings-graph.component.ts [hunk 3]
```diff
@@ -663,7 +663,7 @@ export class ReadingsGraphComponent implements OnDestroy {
 
   create3DGraph(readings: any, ts: any) {
     readings = orderBy(readings, [reading => reading.key.toLowerCase()], ['asc']);
-    const timestamps = ts.map((t: any) => this.dateFormatter.transform(t, 'HH:mm:ss:SSS'));
+    const timestamps = ts.map((t: any) => this.dateFormatter.transform(t, 'HH:mm:ss.SSS'));
```
Comment (mshariq-nerd) [inline src/app/components/core/asset-readings/readings-graph/readings-graph.component.ts:666]: Missing `YYYY-MM-DD` with time format

**Review by mshariq-nerd — CHANGES_REQUESTED** [review]

File: src/app/components/core/asset-readings/readings-graph/readings-graph.component.ts (hunk 3 above)
Comment (praveen-garg) [reply src/app/components/core/asset-readings/readings-graph/readings-graph.component.ts:666]: We should probably not show YMD thing for 3D graph!

Comment (mshariq-nerd) [reply src/app/components/core/asset-readings/readings-graph/readings-graph.component.ts:666]: Agree!

**Comment by praveen-garg** [comment]
Also if data is more than 1 day we should show day on x axis! Imagine user viewing data for 3rd or 4th day while rendered for a week.

File: src/app/components/core/asset-readings/readings-graph/readings-graph.component.ts (hunk 1 above)
Comment (praveen-garg) [reply src/app/components/core/asset-readings/readings-graph/readings-graph.component.ts:628]: We will add a new Jira for this for >1 day data.

**Review by praveen-garg — APPROVED** [review]

**Review by mshariq-nerd — APPROVED** [review]

**Pull request merged**